data/users/externaloidc.py

```diff
@@ -37,10 +39,37 @@ def verify_credentials(self, username_or_email, password):
         """
         Unsupported to login via username/email and password
         """
-        return (
-            None,
-            f"Unsupported login option. Please sign in with the configured {self._federated_service} provider",
-        )
+        if not password:
+            return (None, "Anonymous binding not allowed.")
+
+        if not username_or_email:
+            return (None, "Missing username or email.")
+
+        for service in app.oauth_login.services:
+            if isinstance(service, OIDCLoginService):
+                try:
+                    response = service.password_grant_for_login(username_or_email, password)
+
+                    if response is None:
+                        return (None, "External OIDC Group Sync: Got no user info")
+
+                    user_info = service.get_user_info(
+                        service._http_client, response["access_token"]
+                    )
+
+                    return (
+                        UserInformation(
+                            username=user_info.get("preferred_username"),
```

Comment:
`preferred_username` is not always present. Check the logic in `exchange_code_for_login` to see how it deals with this. If possible you could reuse part of that logic